libs/repositories/src/UsdRepository/UsdRepository.ts

```diff
@@ -20,11 +22,30 @@ export interface PricePoint {
 }
 
 export interface UsdRepository {
-  getUsdPrice(chainId: number, tokenAddress: string): Promise<number | null>;
+  getUsdPrice(
+    chainId: SupportedChainId,
+    tokenAddress: string
+  ): Promise<number | null>;
 
   getUsdPrices(
-    chainId: number,
+    chainId: SupportedChainId,
     tokenAddress: string,
     priceStrategy: PriceStrategy
   ): Promise<PricePoint[] | null>;
 }
+
+export class UsdRepositoryNoop implements UsdRepository {
```

Comment:
Why a noop repository?

Reply:
Its convenient. Both methods return null if they don't know how to get the data. For example, its an unknown token for coingecko.

This is just convenient for extending it and only implement the getUsdPrice for the cow native price implementation, but leave the other method returning always `null`

Reply:
But, returning null in some cases is not the same as a noop operation, IMO.

Maybe I'm not familiar with the class noop pattern, or it's a matter of naming. Don't what to name this, though.

Reply:
Because `null` means here, "I don't know", `noop" is how a repository that complies with the interface but it never know how to get USD prices, or USD price points

For me, it was just convenient primitive to have for testing and for defining repost with a "Default implementation" which just replies "I don't know" (but can be overrides)

anyways, if this confuses you, I will delete it so we don't have to give it a name, and implement the "I don't know" methods splicitly wherever I need them